src/client/js/components/circle-chart.js

```diff
@@ -0,0 +1,244 @@
+        break
+      case 'show-percentage-for':
+        this.showPercentageFor = newValue
+        break
+      case 'title':
+        this.title = newValue
+        break
+      default:
+        console.warning(`Unhandled attribute: ${name}`)
+    }
+  }
+
+  composeCircles () {
+    let sliceOffset = 0
+    return `
+      ${this.data.reduce((acc, curr) => {
+        const percentage = calcPercentage(this.total, curr.count)
+        const innerRadius = this.isDonut ? 0.85 : 0
+        const strokeLength = CHART_CIRCUMFERENCE * percentage
+
+        const circle = `
+          <circle
+            stroke='${curr.color}'
+            stroke-dasharray='${strokeLength} ${CHART_CIRCUMFERENCE}'
+            stroke-dashoffset='${-1 * CHART_CIRCUMFERENCE * sliceOffset}'
+            stroke-width='${CHART_DIAMETER * (1 - innerRadius)}%'
+          ></circle>
+        `
+        acc.push(circle)
+        sliceOffset += percentage
+
+        return acc
+      }, []).join('')}
+    `
+  }
+
+  createChartLabels () {
+    return `
+      ${this.title !== ''
+        ? `<strong class='circle-chart-title'>${this.title}</strong>`
+        : ''}
+      ${this.data.map(({ name, color }) => (
+        `<label class='circle-chart-label' style='color: ${color}'>${name}</label>`
+      )).join('')}
+    `
+  }
+
+  createCircleLabel () {
+    const relevantItem = this.data.find(d => d.key === this.showPercentageFor)
+    if (!relevantItem) {
+      return ''
+    }
+    const percentage = calcPercentage(this.total, relevantItem.count)
+    return `
+      <text
+        dy='${CHART_RADIUS * 0.15}'
+        fill='${relevantItem.color}'
+        font-size='${CHART_RADIUS * 0.4}'
+        font-size='50'
+        x='${CHART_RADIUS}'
+        y='${CHART_RADIUS}'
+      >
+          ${Math.round(percentage * 100)}%
+      </text>
+    `
+  }
+
+  createChart () {
+    // Create SVG with circles
+    const circles = this.composeCircles()
+    this.svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg')
+    this.svg.setAttribute('viewBox', `0 0 ${CHART_DIAMETER} ${CHART_DIAMETER}`)
+    this.svg.innerHTML = `
+      ${circles}
+      ${this.createCircleLabel()}
+    `
+
+    // Create labels
+    this.labels = document.createElement('figcaption')
+    this.labels.innerHTML = this.createChartLabels()
+
+    // Add chart elements to DOM
+    this.chartElement.append(this.svg)
+    this.chartElement.append(this.labels)
+  }
+
+  updateChart () {
+    this.svg.innerHTML = `
+      ${this.composeCircles()}
+      ${this.createCircleLabel()}
+    `
+    this.labels.innerHTML = this.createChartLabels()
+  }
+
+  createOrUpdateChart () {
+    if (this.updateTimeout) {
+      clearTimeout(this.updateTimeout)
+    }
+
+    if (!this.data) {
+      return
+    }
+
+    this.total = this.data.reduce((acc, curr) => acc + curr.count, 0)
+
+    this.chartElement.classList.add('updating')
+    this.updateTimeout = setTimeout(() => {
```

Comment:
I'm confused at the need for a timeout?  Is there a better way?  Happy to help here if needed.